### PR TITLE
Minor docs updates for Swift public beta

### PR DESCRIPTION
**Pull request by felicitymay**
This pull requests adds a few small changes that are required for the public beta release of Swift analysis.

It should not be merged until the following associated PR is ready to merge:

- https://github.com/github/codeql/pull/12940

I'll be raising separate PRs for the updates needed in internal repositories.

The test failures are expected until the related PRs are merged.

### CODE REVIEW

File: docs/codeql/writing-codeql-queries/creating-path-queries.rst [hunk 1]
```diff
@@ -30,7 +30,7 @@ For more language-specific information on analyzing data flow, see:
 - ":ref:`Analyzing data flow in JavaScript/TypeScript <analyzing-data-flow-in-javascript-and-typescript>`"
 - ":ref:`Analyzing data flow in Python <analyzing-data-flow-in-python>`"
 - ":ref:`Analyzing data flow in Ruby <analyzing-data-flow-in-ruby>`"
-
+- ":ref:`Analyzing data flow in Swift <analyzing-data-flow-in-swift>`"
 
 Path query examples
```
Comment (felicitymay) [inline docs/codeql/writing-codeql-queries/creating-path-queries.rst:35]: @geoffw0 - I'm wondering if Swift needs to be added to the list of languages that need you to explicitly import the data flow library or not. See: [Constructing a path query](https://codeql.github.com/docs/writing-codeql-queries/creating-path-queries/#constructing-a-path-query).

Comment (geoffw0) [reply docs/codeql/writing-codeql-queries/creating-path-queries.rst:35]: Yes, it appears the dataflow library does need to be imported in Swift (for both old and new `ConfigSig` style dataflow queries).  It doesn't like:
```
import semmle.code.swift.dataflow.DataFlow
import DataFlow::PathGraph
```
though, it wants:
```
import codeql.swift.dataflow.DataFlow
import DataFlow::PathGraph
```
(I guess this difference is covered by the phrasing "such as")

Comment (felicitymay) [reply docs/codeql/writing-codeql-queries/creating-path-queries.rst:35]: Updated.

**Review by geoffw0 — APPROVED** [review]
Looks great!  (assuming CI checks pass)

**Pull request merged**